kolibri/plugins/learn/viewsets.py

```diff
@@ -216,10 +222,10 @@ def consolidate(self, items, queryset):
                     "closed": None,
                     "started": False,
                 }
-            exam["missing_resource"] = any(
-                question["exercise_id"] not in available_exam_ids
-                for question in exam.get("question_sources")
-            )
+            missing_resource = False
+            for question_source in exam["question_sources"]:
+                missing_resource = not question_source["exercise_id"] in contentnode_map
```

Comment:
We can shortcut the evaluation here (so it's parallel to the previous `any`) by breaking out of the loop in this case.

So could change this to:
```
            for question_source in exam["question_sources"]:
                if question_source["exercise_id"] not in contentnode_map:
                    missing_resource = True
                    break
```

Note also the slightly more idiomatic `not in` syntax rather than negating an `in` expression.

Reply:
Right, when the `contentnode_map` doesn't contain the `exercise_id` of one question, the `missing_resource` will continue to be false. So no need to check every exam question!